### PR TITLE
Prevent padding from exceeding bounds on Android

### PR DESCRIPTION
**Pull request by naftalibeder**
This is essentially a port of the logic that @mfazekas created in #1057 for iOS, which prevents a crash in the event of an 'illegal' layout configuration involving map padding along one axis that exceeds the dimensions of the map in that direction.

For example, if the top and bottom padding exceed the height of the map, this allows it to make a decent guess instead of simply crashing.

More specifically, the logic mirrors what I suggested [here](https://github.com/react-native-mapbox-gl/maps/pull/1057#discussion_r501996207), because I was still having problems with the exact logic in that PR. My justification for handling the error in this way is also detailed in that discussion.

### CODE REVIEW

**Review by ferdicus — COMMENTED** [review]
Looks sane to me comparing to the iOS solution. 

With the different philosophies on gracefully saving when it's exceeding - you guys have to duke that one out between yourselves :)

Once you come to a resolution I'd like it if both solutions were consistent. 

Here for example it reads `overflow` in the iOS solution it's `extra`

Thanks in advance

File: android/rctmgl/src/main/java/com/mapbox/rctmgl/components/camera/CameraStop.java [hunk 1]
```diff
@@ -184,6 +191,27 @@ public static CameraStop fromReadableMap(Context context, @NonNull ReadableMap r
...
+        int top = padding[1];
+        int right = padding[2];
+        int bottom = padding[3];
+        int height = mapView.getHeight();
+        int width = mapView.getWidth();
+
+        if (top + bottom >= height) {
+            double overflow =  top + bottom - height;
+            top -= overflow / 2.0 + 1;
+            bottom -= overflow / 2.0 + 1;
```
Comment (mfazekas) [inline android/rctmgl/src/main/java/com/mapbox/rctmgl/components/camera/CameraStop.java:205]:
```suggestion
            double overflow =  top + bottom - height + 1.0;
            top -= overflow * top / (top + bottom);
            bottom -= overflow * bottom / (top + bottom);
```

File: android/rctmgl/src/main/java/com/mapbox/rctmgl/components/camera/CameraStop.java [hunk 2]
```diff
@@ -184,6 +191,27 @@ public static CameraStop fromReadableMap(Context context, @NonNull ReadableMap r
...
+        }
+        if (left + right >= width) {
+            double overflow =  left + right - width;
+            left -= overflow / 2.0 + 1;
+            right -= overflow / 2.0 + 1;
```
Comment (mfazekas) [inline android/rctmgl/src/main/java/com/mapbox/rctmgl/components/camera/CameraStop.java:210]:
```suggestion
            double overflow =  left + right - width + 1.0;
            left -= overflow * left / (left + right);
            right -= overflow * right / (left + right);
```

**Review by mfazekas — CHANGES_REQUESTED** [review]
👍 

let sync the alg to the one used in ios

**Review by mfazekas — APPROVED** [review]

**Pull request closed without merging**